### PR TITLE
Remove strtok, strtok_s, wcstok_s PAL APIs

### PR DESCRIPTION
**Pull request by jkoritzinsky**
These APIs were used extremely rarely throughout CoreCLR (one usage each), so replace the usages with platform-native APIs (strtok_r for strtok and strtok_s) or use SString to hand-roll the one usage (for wcstok_s).

### CODE REVIEW

File: src/coreclr/vm/profilinghelper.cpp [hunk 1]
```diff
@@ -806,23 +806,28 @@ HRESULT ProfilingAPIUtility::AttemptLoadProfilerList()
...
+        SString::Iterator pathEnd = sectionStart;
+        if (!profilerList.Find(pathEnd, W('=')) || pathEnd > sectionEnd)
+        {
+            ProfilingAPIUtility::LogProfError(IDS_E_PROF_BAD_PATH);
+            storedHr = E_FAIL;
+            continue;
+        }
+        SString::Iterator clsidStart = pathEnd + 1;
+        profilerList.Replace(pathEnd, W('\0'));
+        profilerList.Replace(sectionEnd, W('\0'));
```
Comment (jkotas) [inline src/coreclr/vm/profilinghelper.cpp:825]: This hack with inserting null terminators does not look great.

If you want to use SString, can you create the proper substring using the SString constructor that takes start & end?

File: src/coreclr/vm/profilinghelper.cpp [hunk 2]
```diff
@@ -806,23 +806,27 @@ HRESULT ProfilingAPIUtility::AttemptLoadProfilerList()
...
+        {
+            ProfilingAPIUtility::LogProfError(IDS_E_PROF_BAD_PATH);
+            storedHr = E_FAIL;
+            continue;
+        }
+        SString::Iterator clsidStart = pathEnd + 1;
 
+        PathString path{profilerList, sectionStart, pathEnd};
+        StackSString clsidString{profilerList, clsidStart, sectionEnd};
+        NewArrayHolder<WCHAR> clsidStringRaw = clsidString.GetCopyOfUnicodeString();
```
Comment (jkotas) [inline src/coreclr/vm/profilinghelper.cpp:827]:
```suggestion
```

File: src/coreclr/vm/profilinghelper.cpp [hunk 3]
```diff
@@ -806,23 +806,27 @@ HRESULT ProfilingAPIUtility::AttemptLoadProfilerList()
...
         CLSID clsid;
-        hr = ProfilingAPIUtility::ProfilerCLSIDFromString(currentGuid, &clsid);
+        hr = ProfilingAPIUtility::ProfilerCLSIDFromString(clsidStringRaw, &clsid);
```
Comment (jkotas) [inline src/coreclr/vm/profilinghelper.cpp:829]:
```suggestion
        hr = ProfilingAPIUtility::ProfilerCLSIDFromString(clsidString.GetUnicode(), &clsid);
```
?

**Review by jkotas — APPROVED** [review]
Thanks

**Comment by jkoritzinsky** [comment]
@dotnet-policy-service rerun

**Pull request merged**